src/api/authereum/deposit.js

```diff
@@ -0,0 +1,31 @@
+const { post } = require('request-promise')
+const validateAssertions = require('../../lib/validators/validateAssertions')
+
+module.exports = async (dvf, token, amount, nonce, signature) => {
+  validateAssertions(dvf, { amount, token })
+  const tempVaultId = dvf.config.DVF.tempStarkVaultId || '1'
+  const depositAmount = dvf.util.prepareDepositAmount(amount, token)
+  const starkVaultId = await dvf.getVaultId(token, nonce, signature)
+  const starkDeposit = await dvf.stark.authereum.createSignedTransfer(
+    token,
+    depositAmount,
+    tempVaultId,
+    starkVaultId
+  )
```

Comment:
Just a general comment - it seems like we are duplicating alot of code for each of the Authereum endpoints, when its mainly the signing that is different.

Maybe we should refactor wallet support to avoid duplicating code and just have the logic for signing handled in one place

Reply:
I have some ideas in mind for refactoring but kept the approach the same as ledger for simplicity and I tried reusing as much code as possible